src/core/public/application/application_service.tsx

```diff
@@ -105,23 +153,41 @@ export class ApplicationService {
     }
 
     // Disable registration of new applications
-    this.apps$.complete();
-    this.legacyApps$.complete();
+    this.started = true;
 
     const legacyMode = injectedMetadata.getLegacyMode();
     const currentAppId$ = new BehaviorSubject<string | undefined>(undefined);
-    const { availableApps, availableLegacyApps, capabilities } = await this.capabilities.start({
-      apps: new Map([...this.apps$.value].map(([id, { app }]) => [id, app])),
-      legacyApps: this.legacyApps$.value,
+    const { availableApps, capabilities } = await this.capabilities.start({
+      apps: new Map(
+        [...this.apps].map(([id, appBox]) => [
+          id,
+          (isLegacyApp(appBox) ? appBox : appBox.app) as App | LegacyApp,
+        ])
+      ),
       injectedMetadata,
     });
 
     // Only setup history if we're not in legacy mode
     const history = legacyMode ? null : createBrowserHistory({ basename: http.basePath.get() });
 
+    const availableApps$ = new BehaviorSubject<ReadonlyMap<string, App | LegacyApp>>(availableApps);
+
+    combineLatest([
+      of(availableApps),
```

Comment:
Right here I found it a bit confusing that there are two variables. One called `availableApps` and another `availableApps$`. Can we create two more distinct names?